core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -147,6 +149,61 @@ public Address getFieldValue(MasterInfo o) {
   /** Path level properties. */
   private PathProperties mPathProperties;
 
+  /** Persisted state for MetaMaster. */
+  private State mState;
+
+  /**
+   * Journaled state for MetaMaster.
+   */
+  @NotThreadSafe
+  public static final class State implements alluxio.master.journal.Journaled {
+    /** A unique ID to identify the cluster. */
+    private String mClusterID;
+
+    /**
+     * @return the cluster ID
+     */
+    public String getClusterID() {
+      return mClusterID;
+    }
+
+    @Override
+    public CheckpointName getCheckpointName() {
+      return CheckpointName.CLUSTER_ID;
+    }
+
+    @Override
+    public boolean processJournalEntry(Journal.JournalEntry entry) {
+      if (entry.hasClusterId()) {
+        mClusterID = entry.getClusterId();
+        return true;
+      }
+      return false;
+    }
+
+    /**
+     * @param ctx the journal context
+     * @param entry the clusterID journal entry
+     */
+    public void applyAndJournal(java.util.function.Supplier<JournalContext> ctx, String entry) {
+      applyAndJournal(ctx, Journal.JournalEntry.newBuilder().setClusterId(entry).build());
+    }
+
+    @Override
+    public void resetState() {
+      mClusterID = null;
```

Comment:
Should we have a constant id which represents the reset one (instead of using `null`)?

Reply:
sgtm